core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1198,9 +1198,14 @@ public static function generateFrontendUrl(array $arrRow, $strParams=null, $strF
 		$objRouter = System::getContainer()->get('router');
 		$strUrl = $objRouter->generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, array(RouteObjectInterface::CONTENT_OBJECT => $page, 'parameters' => $strParams));
 
-		// Remove path from absolute URLs
-		if (0 === strncmp($strUrl, '/', 1))
+		if (0 === strncmp($strUrl, '//', 2))
 		{
+			// Add the scheme to network paths (see #2262)
+			$strUrl = ($page->rootUseSSL ? 'https:' : 'http:') . $strUrl;
```

Comment:
This will create `http://` URLs even if the current request is `https`. Both might be valid.

Reply:
This will create HTTP URLs if the page settings say so. Perfectly fine IMHO. We must not assume HTTPS if the checkbox is not enabled.

Reply:
As discussed at the developer's meeting, we want to change the logic this way (see #2323).